wrapanapi/systems/msazure.py

```diff
@@ -771,17 +789,17 @@ def remove_nics_by_search(self, nic_template, resource_group=None):
             operation.wait()
             self.logger.info('"%s" nic removed', nic)
             results.append((nic, operation.status()))
-        if not results:
+        if nic_template and not results:
```

Comment:
This message should be logged regardless of whether `nic_template` was set.

The logging should handle `nic_template` being empty and still writing a sensible message.

Reply:
Removed checking of nic_template availability and throwing a general mssg if no results though its nic_template or all.